src/server/utils/wallets/createSmartWallet.ts

```diff
@@ -0,0 +1,152 @@
+import { defineChain, type Address } from "thirdweb";
+import { smartWallet, type Account } from "thirdweb/wallets";
+import { createWalletDetails } from "../../../db/wallets/createWalletDetails";
+import { WalletType } from "../../../schema/wallet";
+import { thirdwebClient } from "../../../utils/sdk";
+import { splitAwsKmsArn } from "./awsKmsArn";
+import {
+  createAwsKmsWallet,
+  type CreateAwsKmsWalletParams,
+} from "./createAwsKmsWallet";
+import {
+  createGcpKmsWallet,
+  type CreateGcpKmsWalletParams,
+} from "./createGcpKmsWallet";
+import { createLocalWallet } from "./createLocalWallet";
+import { getAwsKmsAccount } from "./getAwsKmsAccount";
+import { getGcpKmsAccount } from "./getGcpKmsAccount";
+
+const createSmartWallet = async ({
```

Comment:
Agreed this is confusing, I like your `generate` suggestion. It's still a little unclear because for AWS and GCP, generate would still have side effect (interactions with AWS and GCP).